libsolidity/analysis/OverrideChecker.cpp

```diff
@@ -584,35 +584,35 @@ void OverrideChecker::checkOverride(OverrideProxy const& _overriding, OverridePr
 				"Overridden " + _overriding.astNodeName() + " is here:"
 			);
 
-		// This is only relevant for a function overriding a function.
-		if (_overriding.isFunction())
-		{
-			// Stricter mutability is always okay except when super is Payable
-			if ((
+		// Stricter mutability is always okay except when super is Payable
```

Comment:
I was thinking that you don't actually need to change this whole block at all other than adding the `|| isVariable` part.
Is there an advantage of having `if (_overriding.unimplemented() && !_super.unimplemented())` outside the if-block?

The `if` would be slightly easier to read if we just do a small change